tortoise/threshold.go

```diff
@@ -84,16 +85,52 @@ func computeExpectedWeight(weights map[types.EpochID]weight, from, to types.Laye
 
 func computeLocalThreshold(config Config, epochWeight map[types.EpochID]weight, lid types.LayerID) weight {
 	threshold := weightFromUint64(0)
-	// TODO(dshulyak) expected weight for local threshold should be based on last layer.
 	threshold = threshold.add(epochWeight[lid.GetEpoch()])
 	threshold = threshold.fraction(config.LocalThreshold)
 	return threshold
 }
 
-func computeGlobalThreshold(config Config, epochWeight map[types.EpochID]weight, target, last types.LayerID) weight {
+func computeThresholdForLayers(config Config, epochWeight map[types.EpochID]weight, target, last types.LayerID) weight {
 	expected := computeExpectedWeight(epochWeight, target, last)
 	threshold := weightFromUint64(0)
 	threshold = threshold.add(expected)
 	threshold = threshold.fraction(config.GlobalThreshold)
 	return threshold
 }
+
+func getConfidenceWindow(config Config, state *commonState, tmode mode) types.LayerID {
+	target := state.verified.Add(1)
+	window := state.last
+	if tmode.isFull() && state.last.Difference(target) > config.FullModeRerunWindow {
+		window = target.Add(config.FullModeRerunWindow)
+	} else if tmode.isVerifying() && state.last.Difference(target) > config.VerifyingModeRerunWindow {
+		window = target.Add(config.VerifyingModeRerunWindow)
+	}
```

Comment:
do you plan for this to have a 3rd kind of window for sync?

Reply:
i think sync should use same window, i will rename it. maybe FullModeVerificationWindow

Reply:
basically it should be applied any time when last layer - verified > this window